File: Python/day01/day01.py
# link to source problem: https://adventofcode.com/2022/day/1


def read_file(path):
    result = []
    with open(path, 'r') as file:
        elf_calories = []
        for line in file.readlines():
            if line == '\n':
                result.append(elf_calories)
                elf_calories = []
                continue
            else:
                elf_calories.append(int(line.strip()))
        return result


def task01(parsed_input):
    max_calories = 0
    for item in parsed_input:
        sum_of_calories = sum(item)
        if sum_of_calories > max_calories:
            max_calories = sum_of_calories
    return max_calories


def task02(parsed_input):
    sums_of_calories = []
    for item in parsed_input:
        sums_of_calories.append(sum(item))
    sums_of_calories = sorted(sums_of_calories, reverse=True)
    return sums_of_calories[0] + sums_of_calories[1] + sums_of_calories[2]


prepared_file = read_file('day01.txt')
print(prepared_file)
print(task01(prepared_file))
print(task02(prepared_file))

File: Java/day01/src/day01.java
// link to source problem: https://adventofcode.com/2022/day/1

import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

public class day01 {

    public static ArrayList<ArrayList<Integer>> read_file(String path) {
        ArrayList<ArrayList<Integer>> result = new ArrayList<>();
        try{
            File myInput = new File(path);
            Scanner myReader = new Scanner(myInput);
            ArrayList<Integer> elfCalories = new ArrayList<>();
            while (myReader.hasNextLine()) {
                String line = myReader.nextLine();
                if (line.equals("")) {
                    result.add(elfCalories);
                    elfCalories = new ArrayList<>();
                } else {
                    elfCalories.add(Integer.parseInt(line));
                }
            }
        } catch (FileNotFoundException e) {
            System.out.println("File does not exist.");
            e.printStackTrace();
        }
        return result;
    }

    public static void main(String[] args) {
        ArrayList<ArrayList<Integer>> parsed_file = read_file("src/day01.txt");
        ArrayList<Long> caloriesSums = new ArrayList<>();
        for (ArrayList<Integer> integers : parsed_file) {
            long calSum = 0L;
            for (Integer integer : integers) {
                calSum += integer;
            }
            caloriesSums.add(calSum);
        }
        caloriesSums.sort(Collections.reverseOrder());
        System.out.format("Maximum calories: %d\n", caloriesSums.get(0));
        System.out.format("Top three: %d\n", caloriesSums.get(0) + caloriesSums.get(1) + caloriesSums.get(2));
    }
}